The job **`aarch64-gnu-llvm-21-2`** failed! Check out the build log: [(web)](https://github.com/rust-lang/rust/actions/runs/25196997055/job/73879815126) [(plain enhanced)](https://triage.rust-lang.org/gha-logs/rust-lang/rust/73879815126) [(plain)](https://github.com/rust-lang/rust/commit/d7f14d3eb6b9a99fc18d7d807d218e1379ab3d9a/checks/73879815126/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain
##[endgroup]
Executing "/scripts/stage_2_test_set2.sh"
+ /scripts/stage_2_test_set2.sh
+ '[' 1 == 1 ']'
+ echo 'PR_CI_JOB set; skipping tidy'
+ SKIP_TIDY='--skip tidy'
+ ../x.py --stage 2 test --skip tidy --skip tests --skip coverage-map --skip coverage-run --skip library --skip tidyselftest
PR_CI_JOB set; skipping tidy
##[group]Building bootstrap
    Finished `dev` profile [unoptimized] target(s) in 0.04s
##[endgroup]
---
   Compiling rustc_graphviz v0.0.0 (/checkout/compiler/rustc_graphviz)
error[E0107]: trait takes 0 lifetime arguments but 1 lifetime argument was supplied
   --> compiler/rustc_graphviz/src/tests.rs:104:10
    |
104 | impl<'a> Labeller<'a> for LabelledGraph {
    |          ^^^^^^^^---- help: remove the unnecessary generics
    |          |
    |          expected 0 lifetime arguments
    |
note: trait defined here, with 0 lifetime parameters
---

error[E0107]: trait takes 0 lifetime arguments but 1 lifetime argument was supplied
   --> compiler/rustc_graphviz/src/tests.rs:130:10
    |
130 | impl<'a> Labeller<'a> for LabelledGraphWithEscStrs {
    |          ^^^^^^^^---- help: remove the unnecessary generics
    |          |
    |          expected 0 lifetime arguments
    |
note: trait defined here, with 0 lifetime parameters
---

error[E0107]: trait takes 0 lifetime arguments but 1 lifetime argument was supplied
   --> compiler/rustc_graphviz/src/tests.rs:151:10
    |
151 | impl<'a> GraphWalk<'a> for LabelledGraph {
    |          ^^^^^^^^^---- help: remove the unnecessary generics
    |          |
    |          expected 0 lifetime arguments
    |
note: trait defined here, with 0 lifetime parameters
---

error[E0107]: trait takes 0 lifetime arguments but 1 lifetime argument was supplied
   --> compiler/rustc_graphviz/src/tests.rs:168:10
    |
168 | impl<'a> GraphWalk<'a> for LabelledGraphWithEscStrs {
    |          ^^^^^^^^^---- help: remove the unnecessary generics
    |          |
    |          expected 0 lifetime arguments
    |
note: trait defined here, with 0 lifetime parameters
---

error[E0308]: method not compatible with trait
   --> compiler/rustc_graphviz/src/tests.rs:107:5
    |
107 |     fn graph_id(&'a self) -> Id<'a> {
    |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ lifetime mismatch
    |
    = note: expected signature `fn(&LabelledGraph) -> Id<'_>`
               found signature `fn(&'a LabelledGraph) -> Id<'a>`
note: the anonymous lifetime defined here...
   --> compiler/rustc_graphviz/src/lib.rs:428:17
    |
428 |     fn graph_id(&self) -> Id<'_>;
    |                 ^^^^^
note: ...does not necessarily outlive the lifetime `'a` as defined here
   --> compiler/rustc_graphviz/src/tests.rs:104:6
    |
104 | impl<'a> Labeller<'a> for LabelledGraph {
    |      ^^

error[E0308]: method not compatible with trait
   --> compiler/rustc_graphviz/src/tests.rs:107:5
    |
107 |     fn graph_id(&'a self) -> Id<'a> {
    |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ lifetime mismatch
    |
    = note: expected signature `fn(&LabelledGraph) -> Id<'_>`
               found signature `fn(&'a LabelledGraph) -> Id<'a>`
note: the lifetime `'a` as defined here...
   --> compiler/rustc_graphviz/src/tests.rs:104:6
    |
104 | impl<'a> Labeller<'a> for LabelledGraph {
    |      ^^
note: ...does not necessarily outlive the anonymous lifetime defined here
   --> compiler/rustc_graphviz/src/lib.rs:428:17
    |
428 |     fn graph_id(&self) -> Id<'_>;
    |                 ^^^^^

error[E0308]: method not compatible with trait
   --> compiler/rustc_graphviz/src/tests.rs:110:5
    |
110 |     fn node_id(&'a self, n: &Node) -> Id<'a> {
    |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ lifetime mismatch
    |
    = note: expected signature `fn(&LabelledGraph, &_) -> Id<'_>`
               found signature `fn(&'a LabelledGraph, &_) -> Id<'a>`
note: the anonymous lifetime defined here...
   --> compiler/rustc_graphviz/src/lib.rs:433:16
    |
433 |     fn node_id(&self, n: &Self::Node) -> Id<'_>;
    |                ^^^^^
note: ...does not necessarily outlive the lifetime `'a` as defined here
   --> compiler/rustc_graphviz/src/tests.rs:104:6
    |
104 | impl<'a> Labeller<'a> for LabelledGraph {
    |      ^^

error[E0308]: method not compatible with trait
   --> compiler/rustc_graphviz/src/tests.rs:110:5
    |
110 |     fn node_id(&'a self, n: &Node) -> Id<'a> {
    |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ lifetime mismatch
    |
    = note: expected signature `fn(&LabelledGraph, &_) -> Id<'_>`
               found signature `fn(&'a LabelledGraph, &_) -> Id<'a>`
note: the lifetime `'a` as defined here...
   --> compiler/rustc_graphviz/src/tests.rs:104:6
    |
104 | impl<'a> Labeller<'a> for LabelledGraph {
    |      ^^
note: ...does not necessarily outlive the anonymous lifetime defined here
   --> compiler/rustc_graphviz/src/lib.rs:433:16
    |
433 |     fn node_id(&self, n: &Self::Node) -> Id<'_>;
    |                ^^^^^

error[E0308]: method not compatible with trait
   --> compiler/rustc_graphviz/src/tests.rs:113:5
    |
113 |     fn node_label(&'a self, n: &Node) -> LabelText<'a> {
    |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ lifetime mismatch
    |
    = note: expected signature `fn(&LabelledGraph, &_) -> LabelText<'_>`
               found signature `fn(&'a LabelledGraph, &_) -> LabelText<'a>`
note: the anonymous lifetime defined here...
   --> compiler/rustc_graphviz/src/lib.rs:446:19
    |
446 |     fn node_label(&self, n: &Self::Node) -> LabelText<'_> {
    |                   ^^^^^
note: ...does not necessarily outlive the lifetime `'a` as defined here
   --> compiler/rustc_graphviz/src/tests.rs:104:6
    |
104 | impl<'a> Labeller<'a> for LabelledGraph {
    |      ^^

error[E0308]: method not compatible with trait
   --> compiler/rustc_graphviz/src/tests.rs:113:5
    |
113 |     fn node_label(&'a self, n: &Node) -> LabelText<'a> {
    |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ lifetime mismatch
    |
    = note: expected signature `fn(&LabelledGraph, &_) -> LabelText<'_>`
               found signature `fn(&'a LabelledGraph, &_) -> LabelText<'a>`
note: the lifetime `'a` as defined here...
   --> compiler/rustc_graphviz/src/tests.rs:104:6
    |
104 | impl<'a> Labeller<'a> for LabelledGraph {
    |      ^^
note: ...does not necessarily outlive the anonymous lifetime defined here
   --> compiler/rustc_graphviz/src/lib.rs:446:19
    |
446 |     fn node_label(&self, n: &Self::Node) -> LabelText<'_> {
    |                   ^^^^^

error[E0308]: method not compatible with trait
   --> compiler/rustc_graphviz/src/tests.rs:119:5
    |
119 |     fn edge_label(&'a self, e: &&'a Edge) -> LabelText<'a> {
    |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ lifetime mismatch
    |
    = note: expected signature `fn(&LabelledGraph, &&Edge) -> LabelText<'_>`
               found signature `fn(&'a LabelledGraph, &&'a Edge) -> LabelText<'a>`
note: the anonymous lifetime defined here...
   --> compiler/rustc_graphviz/src/lib.rs:453:19
    |
453 |     fn edge_label(&self, _e: &Self::Edge) -> LabelText<'_> {
    |                   ^^^^^
note: ...does not necessarily outlive the lifetime `'a` as defined here
   --> compiler/rustc_graphviz/src/tests.rs:104:6
    |
104 | impl<'a> Labeller<'a> for LabelledGraph {
    |      ^^

error[E0308]: method not compatible with trait
   --> compiler/rustc_graphviz/src/tests.rs:119:5
    |
119 |     fn edge_label(&'a self, e: &&'a Edge) -> LabelText<'a> {
    |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ lifetime mismatch
    |
    = note: expected signature `fn(&LabelledGraph, &&Edge) -> LabelText<'_>`
               found signature `fn(&'a LabelledGraph, &&'a Edge) -> LabelText<'a>`
note: the lifetime `'a` as defined here...
   --> compiler/rustc_graphviz/src/tests.rs:104:6
    |
104 | impl<'a> Labeller<'a> for LabelledGraph {
    |      ^^
note: ...does not necessarily outlive the anonymous lifetime defined here
   --> compiler/rustc_graphviz/src/lib.rs:453:19
    |
453 |     fn edge_label(&self, _e: &Self::Edge) -> LabelText<'_> {
    |                   ^^^^^

error[E0308]: method not compatible with trait
   --> compiler/rustc_graphviz/src/tests.rs:122:5
    |
122 |     fn node_style(&'a self, n: &Node) -> Style {
    |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ lifetime mismatch
    |
    = note: expected signature `fn(&LabelledGraph, &_) -> Style`
               found signature `fn(&'a LabelledGraph, &_) -> Style`
note: the anonymous lifetime defined here...
   --> compiler/rustc_graphviz/src/lib.rs:458:19
    |
458 |     fn node_style(&self, _n: &Self::Node) -> Style {
    |                   ^^^^^
note: ...does not necessarily outlive the lifetime `'a` as defined here
   --> compiler/rustc_graphviz/src/tests.rs:104:6
    |
104 | impl<'a> Labeller<'a> for LabelledGraph {
    |      ^^

error[E0308]: method not compatible with trait
   --> compiler/rustc_graphviz/src/tests.rs:125:5
    |
125 |     fn edge_style(&'a self, e: &&'a Edge) -> Style {
    |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ lifetime mismatch
    |
    = note: expected signature `fn(&LabelledGraph, &&Edge) -> Style`
               found signature `fn(&'a LabelledGraph, &&'a Edge) -> Style`
note: the anonymous lifetime defined here...
   --> compiler/rustc_graphviz/src/lib.rs:463:19
    |
463 |     fn edge_style(&self, _e: &Self::Edge) -> Style {
    |                   ^^^^^
note: ...does not necessarily outlive the lifetime `'a` as defined here
   --> compiler/rustc_graphviz/src/tests.rs:104:6
    |
104 | impl<'a> Labeller<'a> for LabelledGraph {
    |      ^^

error[E0207]: the lifetime parameter `'a` is not constrained by the impl trait, self type, or predicates
   --> compiler/rustc_graphviz/src/tests.rs:104:6
    |
104 | impl<'a> Labeller<'a> for LabelledGraph {
    |      ^^ unconstrained lifetime parameter

error[E0308]: method not compatible with trait
   --> compiler/rustc_graphviz/src/tests.rs:133:5
    |
133 |     fn graph_id(&'a self) -> Id<'a> {
    |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ lifetime mismatch
    |
    = note: expected signature `fn(&LabelledGraphWithEscStrs) -> Id<'_>`
               found signature `fn(&'a LabelledGraphWithEscStrs) -> Id<'a>`
note: the anonymous lifetime defined here...
   --> compiler/rustc_graphviz/src/lib.rs:428:17
    |
428 |     fn graph_id(&self) -> Id<'_>;
    |                 ^^^^^
note: ...does not necessarily outlive the lifetime `'a` as defined here
   --> compiler/rustc_graphviz/src/tests.rs:130:6
    |
130 | impl<'a> Labeller<'a> for LabelledGraphWithEscStrs {
    |      ^^

error[E0308]: method not compatible with trait
   --> compiler/rustc_graphviz/src/tests.rs:133:5
    |
133 |     fn graph_id(&'a self) -> Id<'a> {
    |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ lifetime mismatch
    |
    = note: expected signature `fn(&LabelledGraphWithEscStrs) -> Id<'_>`
               found signature `fn(&'a LabelledGraphWithEscStrs) -> Id<'a>`
note: the lifetime `'a` as defined here...
   --> compiler/rustc_graphviz/src/tests.rs:130:6
    |
130 | impl<'a> Labeller<'a> for LabelledGraphWithEscStrs {
    |      ^^
note: ...does not necessarily outlive the anonymous lifetime defined here
   --> compiler/rustc_graphviz/src/lib.rs:428:17
    |
428 |     fn graph_id(&self) -> Id<'_>;
    |                 ^^^^^

error[E0308]: method not compatible with trait
   --> compiler/rustc_graphviz/src/tests.rs:136:5
    |
136 |     fn node_id(&'a self, n: &Node) -> Id<'a> {
    |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ lifetime mismatch
    |
    = note: expected signature `fn(&LabelledGraphWithEscStrs, &_) -> Id<'_>`
               found signature `fn(&'a LabelledGraphWithEscStrs, &_) -> Id<'a>`
note: the anonymous lifetime defined here...
   --> compiler/rustc_graphviz/src/lib.rs:433:16
    |
433 |     fn node_id(&self, n: &Self::Node) -> Id<'_>;
    |                ^^^^^
note: ...does not necessarily outlive the lifetime `'a` as defined here
   --> compiler/rustc_graphviz/src/tests.rs:130:6
    |
130 | impl<'a> Labeller<'a> for LabelledGraphWithEscStrs {
    |      ^^

error[E0308]: method not compatible with trait
   --> compiler/rustc_graphviz/src/tests.rs:136:5
    |
136 |     fn node_id(&'a self, n: &Node) -> Id<'a> {
    |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ lifetime mismatch
    |
    = note: expected signature `fn(&LabelledGraphWithEscStrs, &_) -> Id<'_>`
               found signature `fn(&'a LabelledGraphWithEscStrs, &_) -> Id<'a>`
note: the lifetime `'a` as defined here...
   --> compiler/rustc_graphviz/src/tests.rs:130:6
    |
130 | impl<'a> Labeller<'a> for LabelledGraphWithEscStrs {
    |      ^^
note: ...does not necessarily outlive the anonymous lifetime defined here
   --> compiler/rustc_graphviz/src/lib.rs:433:16
    |
433 |     fn node_id(&self, n: &Self::Node) -> Id<'_>;
    |                ^^^^^

error[E0308]: method not compatible with trait
   --> compiler/rustc_graphviz/src/tests.rs:139:5
    |
139 |     fn node_label(&'a self, n: &Node) -> LabelText<'a> {
    |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ lifetime mismatch
    |
    = note: expected signature `fn(&LabelledGraphWithEscStrs, &_) -> LabelText<'_>`
               found signature `fn(&'a LabelledGraphWithEscStrs, &_) -> LabelText<'a>`
note: the anonymous lifetime defined here...
   --> compiler/rustc_graphviz/src/lib.rs:446:19
    |
446 |     fn node_label(&self, n: &Self::Node) -> LabelText<'_> {
    |                   ^^^^^
note: ...does not necessarily outlive the lifetime `'a` as defined here
   --> compiler/rustc_graphviz/src/tests.rs:130:6
    |
130 | impl<'a> Labeller<'a> for LabelledGraphWithEscStrs {
    |      ^^

error[E0308]: method not compatible with trait
   --> compiler/rustc_graphviz/src/tests.rs:139:5
    |
139 |     fn node_label(&'a self, n: &Node) -> LabelText<'a> {
    |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ lifetime mismatch
    |
    = note: expected signature `fn(&LabelledGraphWithEscStrs, &_) -> LabelText<'_>`
               found signature `fn(&'a LabelledGraphWithEscStrs, &_) -> LabelText<'a>`
note: the lifetime `'a` as defined here...
   --> compiler/rustc_graphviz/src/tests.rs:130:6
    |
130 | impl<'a> Labeller<'a> for LabelledGraphWithEscStrs {
    |      ^^
note: ...does not necessarily outlive the anonymous lifetime defined here
   --> compiler/rustc_graphviz/src/lib.rs:446:19
    |
446 |     fn node_label(&self, n: &Self::Node) -> LabelText<'_> {
    |                   ^^^^^

error[E0308]: method not compatible with trait
   --> compiler/rustc_graphviz/src/tests.rs:144:5
    |
144 |     fn edge_label(&'a self, e: &&'a Edge) -> LabelText<'a> {
    |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ lifetime mismatch
    |
    = note: expected signature `fn(&LabelledGraphWithEscStrs, &&Edge) -> LabelText<'_>`
               found signature `fn(&'a LabelledGraphWithEscStrs, &&'a Edge) -> LabelText<'a>`
note: the anonymous lifetime defined here...
   --> compiler/rustc_graphviz/src/lib.rs:453:19
    |
453 |     fn edge_label(&self, _e: &Self::Edge) -> LabelText<'_> {
    |                   ^^^^^
note: ...does not necessarily outlive the lifetime `'a` as defined here
   --> compiler/rustc_graphviz/src/tests.rs:130:6
    |
130 | impl<'a> Labeller<'a> for LabelledGraphWithEscStrs {
    |      ^^

error[E0308]: method not compatible with trait
   --> compiler/rustc_graphviz/src/tests.rs:144:5
    |
144 |     fn edge_label(&'a self, e: &&'a Edge) -> LabelText<'a> {
    |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ lifetime mismatch
    |
    = note: expected signature `fn(&LabelledGraphWithEscStrs, &&Edge) -> LabelText<'_>`
               found signature `fn(&'a LabelledGraphWithEscStrs, &&'a Edge) -> LabelText<'a>`
note: the lifetime `'a` as defined here...
   --> compiler/rustc_graphviz/src/tests.rs:130:6
    |
130 | impl<'a> Labeller<'a> for LabelledGraphWithEscStrs {
    |      ^^
note: ...does not necessarily outlive the anonymous lifetime defined here
   --> compiler/rustc_graphviz/src/lib.rs:453:19
    |
453 |     fn edge_label(&self, _e: &Self::Edge) -> LabelText<'_> {
    |                   ^^^^^

error[E0207]: the lifetime parameter `'a` is not constrained by the impl trait, self type, or predicates
   --> compiler/rustc_graphviz/src/tests.rs:130:6
    |
130 | impl<'a> Labeller<'a> for LabelledGraphWithEscStrs {
    |      ^^ unconstrained lifetime parameter

error[E0308]: method not compatible with trait
   --> compiler/rustc_graphviz/src/tests.rs:154:5
    |
154 |     fn nodes(&'a self) -> Nodes<'a, Node> {
    |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ lifetime mismatch
    |
    = note: expected signature `fn(&LabelledGraph) -> Cow<'_, _>`
               found signature `fn(&'a LabelledGraph) -> Cow<'a, _>`
note: the anonymous lifetime defined here...
   --> compiler/rustc_graphviz/src/lib.rs:545:14
    |
545 |     fn nodes(&self) -> Nodes<'_, Self::Node>;
    |              ^^^^^
note: ...does not necessarily outlive the lifetime `'a` as defined here
   --> compiler/rustc_graphviz/src/tests.rs:151:6
    |
151 | impl<'a> GraphWalk<'a> for LabelledGraph {
    |      ^^

error[E0308]: method not compatible with trait
   --> compiler/rustc_graphviz/src/tests.rs:154:5
    |
154 |     fn nodes(&'a self) -> Nodes<'a, Node> {
    |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ lifetime mismatch
    |
    = note: expected signature `fn(&LabelledGraph) -> Cow<'_, _>`
               found signature `fn(&'a LabelledGraph) -> Cow<'a, _>`
note: the lifetime `'a` as defined here...
   --> compiler/rustc_graphviz/src/tests.rs:151:6
    |
151 | impl<'a> GraphWalk<'a> for LabelledGraph {
    |      ^^
note: ...does not necessarily outlive the anonymous lifetime defined here
   --> compiler/rustc_graphviz/src/lib.rs:545:14
    |
545 |     fn nodes(&self) -> Nodes<'_, Self::Node>;
    |              ^^^^^

error[E0308]: method not compatible with trait
   --> compiler/rustc_graphviz/src/tests.rs:157:5
    |
157 |     fn edges(&'a self) -> Edges<'a, &'a Edge> {
    |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ lifetime mismatch
    |
    = note: expected signature `fn(&LabelledGraph) -> Cow<'_, [&Edge]>`
               found signature `fn(&'a LabelledGraph) -> Cow<'a, [&'a Edge]>`
note: the anonymous lifetime defined here...
   --> compiler/rustc_graphviz/src/lib.rs:547:14
    |
547 |     fn edges(&self) -> Edges<'_, Self::Edge>;
    |              ^^^^^
note: ...does not necessarily outlive the lifetime `'a` as defined here
   --> compiler/rustc_graphviz/src/tests.rs:151:6
    |
151 | impl<'a> GraphWalk<'a> for LabelledGraph {
    |      ^^

error[E0308]: method not compatible with trait
   --> compiler/rustc_graphviz/src/tests.rs:160:5
    |
160 |     fn source(&'a self, edge: &&'a Edge) -> Node {
    |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ lifetime mismatch
    |
    = note: expected signature `fn(&LabelledGraph, &&Edge) -> _`
               found signature `fn(&'a LabelledGraph, &&'a Edge) -> _`
note: the anonymous lifetime defined here...
   --> compiler/rustc_graphviz/src/lib.rs:549:15
    |
549 |     fn source(&self, edge: &Self::Edge) -> Self::Node;
    |               ^^^^^
note: ...does not necessarily outlive the lifetime `'a` as defined here
   --> compiler/rustc_graphviz/src/tests.rs:151:6
    |
151 | impl<'a> GraphWalk<'a> for LabelledGraph {
    |      ^^

error[E0308]: method not compatible with trait
   --> compiler/rustc_graphviz/src/tests.rs:163:5
    |
163 |     fn target(&'a self, edge: &&'a Edge) -> Node {
    |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ lifetime mismatch
    |
    = note: expected signature `fn(&LabelledGraph, &&Edge) -> _`
               found signature `fn(&'a LabelledGraph, &&'a Edge) -> _`
note: the anonymous lifetime defined here...
   --> compiler/rustc_graphviz/src/lib.rs:551:15
    |
551 |     fn target(&self, edge: &Self::Edge) -> Self::Node;
    |               ^^^^^
note: ...does not necessarily outlive the lifetime `'a` as defined here
   --> compiler/rustc_graphviz/src/tests.rs:151:6
    |
151 | impl<'a> GraphWalk<'a> for LabelledGraph {
    |      ^^

error[E0207]: the lifetime parameter `'a` is not constrained by the impl trait, self type, or predicates
   --> compiler/rustc_graphviz/src/tests.rs:151:6
    |
151 | impl<'a> GraphWalk<'a> for LabelledGraph {
    |      ^^ unconstrained lifetime parameter

error[E0308]: method not compatible with trait
   --> compiler/rustc_graphviz/src/tests.rs:171:5
    |
171 |     fn nodes(&'a self) -> Nodes<'a, Node> {
    |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ lifetime mismatch
    |
    = note: expected signature `fn(&LabelledGraphWithEscStrs) -> Cow<'_, _>`
               found signature `fn(&'a LabelledGraphWithEscStrs) -> Cow<'a, _>`
note: the anonymous lifetime defined here...
   --> compiler/rustc_graphviz/src/lib.rs:545:14
    |
545 |     fn nodes(&self) -> Nodes<'_, Self::Node>;
    |              ^^^^^
note: ...does not necessarily outlive the lifetime `'a` as defined here
   --> compiler/rustc_graphviz/src/tests.rs:168:6
    |
168 | impl<'a> GraphWalk<'a> for LabelledGraphWithEscStrs {
    |      ^^

error[E0308]: method not compatible with trait
   --> compiler/rustc_graphviz/src/tests.rs:171:5
    |
171 |     fn nodes(&'a self) -> Nodes<'a, Node> {
    |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ lifetime mismatch
    |
    = note: expected signature `fn(&LabelledGraphWithEscStrs) -> Cow<'_, _>`
               found signature `fn(&'a LabelledGraphWithEscStrs) -> Cow<'a, _>`
note: the lifetime `'a` as defined here...
   --> compiler/rustc_graphviz/src/tests.rs:168:6
    |
168 | impl<'a> GraphWalk<'a> for LabelledGraphWithEscStrs {
    |      ^^
note: ...does not necessarily outlive the anonymous lifetime defined here
   --> compiler/rustc_graphviz/src/lib.rs:545:14
    |
545 |     fn nodes(&self) -> Nodes<'_, Self::Node>;
    |              ^^^^^

error[E0308]: method not compatible with trait
   --> compiler/rustc_graphviz/src/tests.rs:174:5
    |
174 |     fn edges(&'a self) -> Edges<'a, &'a Edge> {
    |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ lifetime mismatch
    |
```

</details>

--- a/compiler/rustc_borrowck/src/region_infer/values.rs
+++ b/compiler/rustc_borrowck/src/region_infer/values.rs
@@ -426,19 +426,13 @@ impl ToElementIndex<'_> for RegionVid {
 }
 
 impl<'tcx> ToElementIndex<'tcx> for ty::PlaceholderRegion<'tcx> {
-    fn add_to_row<N: Idx>(self, values: &mut RegionValues<'tcx, N>, row: N) -> bool
-    where
-        Self: Into<ty::PlaceholderRegion<'tcx>>,
-    {
+    fn add_to_row<N: Idx>(self, values: &mut RegionValues<'tcx, N>, row: N) -> bool {
         let placeholder: ty::PlaceholderRegion<'tcx> = self.into();
         let index = values.placeholder_indices.lookup_index(placeholder);
         values.placeholders.insert(row, index)
     }
 
-    fn contained_in_row<N: Idx>(self, values: &RegionValues<'tcx, N>, row: N) -> bool
-    where
-        Self: Into<ty::PlaceholderRegion<'tcx>>,
-    {
+    fn contained_in_row<N: Idx>(self, values: &RegionValues<'tcx, N>, row: N) -> bool {
         let placeholder: ty::PlaceholderRegion<'tcx> = self.into();
         let index = values.placeholder_indices.lookup_index(placeholder);
         values.placeholders.contains(row, index)

--- a/compiler/rustc_error_messages/src/lib.rs
+++ b/compiler/rustc_error_messages/src/lib.rs
@@ -268,10 +268,7 @@ pub fn fluent_value_from_str_list_sep_by_and(l: Vec<Cow<'_, str>>) -> FluentValu
         type Args = ();
         type Error = ();
 
-        fn construct(lang: LanguageIdentifier, _args: Self::Args) -> Result<Self, Self::Error>
-        where
-            Self: Sized,
-        {
+        fn construct(lang: LanguageIdentifier, _args: Self::Args) -> Result<Self, Self::Error> {
             let locale = icu_locale_from_unic_langid(lang)
                 .unwrap_or_else(|| rustc_baked_icu_data::supported_locales::EN);
             let list_formatter = icu_list::ListFormatter::try_new_and_unstable(

--- a/compiler/rustc_errors/src/diagnostic.rs
+++ b/compiler/rustc_errors/src/diagnostic.rs
@@ -143,10 +143,7 @@ impl<'a, F: FnOnce(&mut Diag<'_, ()>)> Diagnostic<'a, ()> for DiagDecorator<F> {
 /// Trait implemented by error types. This should not be implemented manually. Instead, use
 /// `#[derive(Subdiagnostic)]` -- see [rustc_macros::Subdiagnostic].
 #[rustc_diagnostic_item = "Subdiagnostic"]
-pub trait Subdiagnostic
-where
-    Self: Sized,
-{
+pub trait Subdiagnostic {
     /// Add a subdiagnostic to an existing diagnostic.
     fn add_to_diag<G: EmissionGuarantee>(self, diag: &mut Diag<'_, G>);
 }

--- a/compiler/rustc_hir/src/intravisit.rs
+++ b/compiler/rustc_hir/src/intravisit.rs
@@ -71,11 +71,6 @@ use rustc_span::{Ident, Span, Symbol};
 
 use crate::hir::*;
 
-pub trait IntoVisitor<'hir> {
-    type Visitor: Visitor<'hir>;
-    fn into_visitor(&self) -> Self::Visitor;
-}
-
 #[derive(Copy, Clone, Debug)]
 pub enum FnKind<'a> {
     /// `#[xxx] pub async/const/extern "Abi" fn foo()`

--- a/compiler/rustc_infer/src/infer/snapshot/undo_log.rs
+++ b/compiler/rustc_infer/src/infer/snapshot/undo_log.rs
@@ -126,7 +126,6 @@ where
 
     fn extend<J>(&mut self, undos: J)
     where
-        Self: Sized,
         J: IntoIterator<Item = T>,
     {
         if self.in_snapshot() {

